### PR TITLE
Constrain NetworkTimeout on IMDS calls from DefaultAzureCredential

### PR DESCRIPTION
**Pull request by schaabs**
Changes to our IMDS discovery model have regressed the perfomance of the `DefaultAzureCredential` on certain platforms where the IMDS endpoint is not available. This change introduces an internal setting to limit the `NetworkTimeout` on our initial IMDS calls. 

The limit is 1 second the same as the original limit which was put on establishing an socket connection in the previous discovery model. If a response is received from the IMDS endpoint the limit is removed.

### CODE REVIEW

File: sdk/identity/Azure.Identity/src/ImdsManagedIdentitySource.cs
Comment (christothes) [inline sdk/identity/Azure.Identity/src/ImdsManagedIdentitySource.cs:103]: Do we actually need to wait for a successful response to unset the timeout? If not, we could elimiante this assignment on each response and make the override look something like this: 

```c#
protected override HttpMessage CreateHttpMessage(Request request)
{
    HttpMessage message = base.CreateHttpMessage(request);
    
    if ( _imdsNetworkTimeout.HasValue)
    {
        message.NetworkTimeout = _imdsNetworkTimeout;
        _imdsNetworkTimeout = null;
    }
    

    return message;
}

```

File: sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs
Comment (christothes) [inline sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs:321]: Let's add a test that validates that the timeout only gets set on the first message.

Comment (schaabs) [reply sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs:321]: I'm still adding tests for this and `DefaultAzureCredential` as well, it's just taking a bit since it requires standing up a listener to force network timeouts etc.

File: sdk/identity/Azure.Identity/src/ImdsManagedIdentitySource.cs
Comment (schaabs) [reply sdk/identity/Azure.Identity/src/ImdsManagedIdentitySource.cs:103]: I was only clearing this when we get a response because we're caching credential instances in the `DefaultAzureCredential`, so if we were to only constrain the first request, subsequent instances of `DefaultAzureCredential` would regress.

File: sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs
Comment (pakrym) [inline sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs:379]: await Task.Delay

File: sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs
Comment (pakrym) [inline sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs:417]: await

File: sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs
Comment (pakrym) [inline sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs:413]: Assert that the test server got 2 requests?

File: eng/Packages.Data.props
Comment (weshaggard) [inline eng/Packages.Data.props:178]: Did you intend to downgrade this?

**Review by christothes — APPROVED** [review]

File: eng/Packages.Data.props
Comment (schaabs) [reply eng/Packages.Data.props:178]: Yes we didn't need 2.2 since I moved this to a more central dependency 2.2 would have required upgrading a bunch of other dependencies.

Comment (weshaggard) [reply eng/Packages.Data.props:178]: I defer to you and @pakrym  on this one.

**Pull request merged**
